static/src/javascripts/projects/commercial/modules/comscore.ts

```diff
@@ -39,35 +34,41 @@ const initOnConsent = () => {
  * Initialise comscore, industry-wide audience tracking
  * https://www.comscore.com/About
  */
-const setupComscore = () => {
+const setupComscore = (): Promise<void> => {
 	if (commercialFeatures.comscore) {
-		return getInitialConsentState()
-			.then((state) => {
-				/* Rule is that comscore can run:
-                - in Tcfv2: Based on consent for comscore
-                - in Australia: Always
-                - in CCPA: If the user hasn't chosen Do Not Sell
-                */
-				const canRunTcfv2 =
-					state.tcfv2 && getConsentFor('comscore', state);
-				const canRunAus = !!state.aus;
-				const canRunCcpa = !!state.ccpa && !state.ccpa.doNotSell;
+		return (
+			getInitialConsentState()
+				.then((state) => {
+					/* Rule is that comscore can run:
+					- in Tcfv2: Based on consent for comscore
+					- in Australia: Always
+					- in CCPA: If the user hasn't chosen Do Not Sell
+					*/
+					const canRunTcfv2 =
+						state.tcfv2 && getConsentFor('comscore', state);
+					const canRunAus = !!state.aus;
+					const canRunCcpa = !!state.ccpa && !state.ccpa.doNotSell;
 
-				if (!(canRunTcfv2 || canRunAus || canRunCcpa)) {
-					return Promise.reject('No consent for comscore');
-				}
-				return initOnConsent();
-			})
-			.catch((e) => {
-				log('commercial', '⚠️ Failed to execute comscore', e);
-			});
+					if (!(canRunTcfv2 || canRunAus || canRunCcpa)) {
+						return Promise.reject('No consent for comscore');
+					}
+					return initOnConsent();
+				})
+				// is there a better way?
```

Comment:
Is there?

Reply:
I think this logic could move to [`getConsentFor`](https://github.com/guardian/consent-management-platform/blob/main/src/getConsentFor.ts).

Reply:
Adding a todo